src/base/libmesh_isnan.c

```diff
@@ -1,34 +1,55 @@
 // The libMesh Finite Element Library.
 // Copyright (C) 2002-2008 Benjamin S. Kirk, John W. Peterson, Roy H. Stogner
-  
+
 // This library is free software; you can redistribute it and/or
 // modify it under the terms of the GNU Lesser General Public
 // License as published by the Free Software Foundation; either
 // version 2.1 of the License, or (at your option) any later version.
-  
+
 // This library is distributed in the hope that it will be useful,
 // but WITHOUT ANY WARRANTY; without even the implied warranty of
 // MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 // Lesser General Public License for more details.
-  
+
 // You should have received a copy of the GNU Lesser General Public
 // License along with this library; if not, write to the Free Software
 // Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
 
- 
 
-// C++ includes
-#include <math.h>
+
+// libMesh includes
 #include "libmesh/libmesh_C_isnan.h"
 
+// C includes
+#include <math.h>
+#include <stdio.h>
+
 #ifdef isnan
-int libmesh_C_isnan_float(float a) { return isnan(a); }
-int libmesh_C_isnan_double(double a) { return isnan(a); }
-int libmesh_C_isnan_longdouble(long double a) { return isnan(a); }
+int libmesh_C_isnan_float(float a) {
+  printf("libmesh_C_isnan_float is deprecated and should not be used!\n");
```

Comment:
Not replicating the libmesh_do_once behavior seems a bit cruel to anybody who's in a situation to get error message spammed here, but hopefully these functions never saw use outside the library itself.